lib/handler/retry-handler.js

```diff
@@ -193,6 +194,19 @@ class RetryHandler {
       this.resume = null
 
       if (statusCode !== 206) {
+        // Abort when status code is not partial content
+        // and payload is already consumed because downstream
+        // will concatenate response from two request wrongly.
```

Comment:
Can you please expand the comment, explaining that anything that is not a 206 implies that a range request cannot be made?